File: ontrack-kdsl/src/main/java/net/nemerosa/ontrack/kdsl/spec/extension/github/ingestion/GitHubIngestionSettings.kt
package net.nemerosa.ontrack.kdsl.spec.extension.github.ingestion

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class GitHubIngestionSettings(
    val token: String,
    val retentionDays: Int,
    val orgProjectPrefix: Boolean,
    val indexationInterval: Int,
    val repositoryIncludes: String,
    val repositoryExcludes: String,
    val issueServiceIdentifier: String,
    val enabled: Boolean,
) {
    fun withToken(token: String) = GitHubIngestionSettings(
        token = token,
        retentionDays = retentionDays,
        orgProjectPrefix = orgProjectPrefix,
        indexationInterval = indexationInterval,
        repositoryIncludes = repositoryIncludes,
        repositoryExcludes = repositoryExcludes,
        issueServiceIdentifier = issueServiceIdentifier,
        enabled = enabled,
    )
}

File: ontrack-web-tests/ontrack/extensions/github/GitHubIngestionSettings.js
import {AbstractSettings} from "@ontrack/AbstractSettings";

export class GitHubIngestionSettings extends AbstractSettings {
    constructor(ontrack) {
        super(ontrack)
    }

    async saveSettings({
                           token,
                           retentionDays,
                           orgProjectPrefix,
                           indexationInterval,
                           repositoryIncludes,
                           repositoryExcludes,
                           issueServiceIdentifier,
                           enabled,
                       }
    ) {
        return this.doSaveSettings({
            id: 'github-ingestion',
            values: {
                token,
                retentionDays,
                orgProjectPrefix,
                indexationInterval,
                repositoryIncludes,
                repositoryExcludes,
                issueServiceIdentifier,
                enabled,
            }
        })
    }
}